cdk/src/open-data-platform/network/dns.ts

```diff
@@ -0,0 +1,60 @@
+// Sets up DNS for the application.
+// Based on https://github.com/BlueConduit/tributary/blob/main/cdk/lib/network/dns.ts
+
+import { Construct } from 'constructs';
+import { Stack } from 'aws-cdk-lib';
+import { EnvType, baseSubdomain, CommonProps } from '../../util';
+import * as route53 from 'aws-cdk-lib/aws-route53';
+import * as certificatemanager from 'aws-cdk-lib/aws-certificatemanager';
+import * as iam from 'aws-cdk-lib/aws-iam';
+
+export class Dns extends Construct {
+  readonly hostedZone: route53.PublicHostedZone;
+  readonly cloudfrontCertificate: certificatemanager.Certificate;
+
+  constructor(scope: Construct, id: string, props: CommonProps) {
+    super(scope, id);
+
+    const { envType } = props;
+
+    const parentDomain = 'blueconduit.com';
+
+    let subdomain = `default.${baseSubdomain}`;
+    if (envType === EnvType.Production) subdomain = baseSubdomain;
+    else if (envType === EnvType.Development) subdomain = `dev.${baseSubdomain}`;
+    else if (envType === EnvType.Sandbox)
+      subdomain = `${process.env.USER ?? 'default'}.sandbox.${baseSubdomain}`;
```

Comment:
I don't think this subdomain structure for dev and sandboxes will work with domain delegation. Except for the root 'blueconduit.com' domain, there can be no overlap between env domains.

Example:

If the root 'blueconduit.com' hosted zone delegates DNS for the 'leadout' subdomain to the prod environment, then even if there's a record that tries to delegate the 'dev.leadout' subdomain to a different env, it will still go to the prod environment

Reply:
We handle this in the saas app because prod is 'app.blueconduit.com', dev is 'dev.blueconduit.com' and our sandboxes are all unique at '${user}.sandbox.blueconduit.com' AND there is no 'sandbox.blueconduit.com' environment

Reply:
proposal for this app is to use something like 'leadout-dev' as the subdomain for the dev env and '${user}.leadout-sandbox' for sandbox envs

Reply:
Thanks for that info. I've updated to follow that pattern.